.github/workflows/pre_release.yaml

```diff
@@ -0,0 +1,36 @@
+name: Canary Release
+
+on:
+  workflow_run:
+    workflows: ["Build and Test"]
```

Comment:
prefer if this was GH trigger based and not based on the workflow names, and causes uneeded coupling

Reply:
This is used for triggering canary-release after completing the build-and-test workflow, when pushing to the main branch, in case that canary has been released but the build and test workflow failed. This is just my concern, because build-and-test workflow often failed after pushing to main, even this commit has passed when sending a PR.

Reply:
I agree with your point that a broken `Build and Test` will result in a stale `edge` artifact, but there are many ways to ensure that doesnt happen.
triggering workflows from UI is common so we dont want this action to get triggered for those cases

Reply:
Thanks, I will change this.